File: Έγγραφα SEM4105/Λύσεις ασκήσεων/03. Τρίτο εργαστήριο/3a/newtonraphson1.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

/*
     Epilisi tis f(x)=0 me f(x)=sin(x)-x**2
     me th methodo Newton-Raphson.
*/


double f(double x);
double df(double x);


int
main()
{    
    double const toler = 1e-8; //     A small constant number

    double x = 1.0;     //    Initial approximation 

    while (fabs(f(x)) > toler) {
	x = x - f(x) / df(x);
    } 

    printf("%s%g\n","The root is approximately ", x);
    printf("%s%g\n", "The function value is ", f(x));

    return 0;
    
}


//    The function
double
f(double x)
{
    return sin(x)-x*x;
}



//     The first derivative 
double
df(double x)
{
    return cos(x) - 2.0 * x;
}

File: Έγγραφα SEM4105/Λύσεις ασκήσεων/03. Τρίτο εργαστήριο/3a/newtonraphson1.f
C     Epilisi tis f(x)=0 me f(x)=sin(x)-x**2
C     me th methodo Newton-Raphson.

      PROGRAM NEWTRA

      DOUBLE PRECISION X, F, DF
C     declare that F, DF are functions
      EXTERNAL F, DF

C     A small constant number
      DOUBLE PRECISION TOLER
      PARAMETER (TOLER = 1D-8)

C     Initial approximation 
      X = 1.0D0

 10   X = X - F(X) / DF(X)
      IF (ABS(F(X)) .GT. TOLER) GOTO 10

      PRINT *, "The root is approximately ", X
      PRINT *, "The function value is ", F(X)

      END

C     The function
      FUNCTION F(X)
      DOUBLE PRECISION X, F

      F = SIN(X)-X**2
      END

C     The first derivative 
      FUNCTION DF(X)
      DOUBLE PRECISION X, DF

      DF = COS(X) - 2.0D0 * X
      END
